### ISSUE TITLE
Import could not be resolved

### ISSUE THREAD

**Issue by TornaxO7** [microsoft/pyright#2257]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
I'm getting an unexpected `Import could not be resolved`.

**To Reproduce**
Create a "project" with the following project tree:

```
├── pyrightconfig.json
└── stuff
    ├── arch
    │   └── linux.py
    │   └── base.py    <-- symlink of stuff/common/base.py
    ├── common
    │   └── base.py
    └── lol
        └── loller.py

4 directories, 4 files
```

Here are the contents of the files:

`pyrightconfig.json`

<details>

```json
{
    "include" : ["./stuff"],
    "pythonVersion" : "3.9",
    "executionEnvironments": [
        {
            "root" : "./stuff/common",
            "extraPaths" : ["./stuff/arch", "./stuff/lol/"]
        },
        {
            "root" : "./stuff/lol/",
            "extraPaths" : ["./stuff/arch", "./stuff/common"],
        },
        {
            "root" : "./stuff/arch/",
            "extraPaths" : ["./stuff/lol", "./stuff/common"]
        }
    ]
}
```

</details>

`stuff/arch/linux.py` (this file includes the unresolved import):

<details>

```python
from common import Base

class Linux:

    def __init__(self) -> None:
        pass

    def greeting(self):
        print("Hello from Linux!")

if __name__ == '__main__':
    base = Base()
    base.greeting()
```

</details>

`stuff/common/base.py`

<details>

```python
 from linux import Linux

class Base:

    def __init__(self) -> None:
        pass

    def greeting(self):
        print("Hello from Base!")


if __name__ == '__main__':
    linux = Linux()
    linux.greeting()
```

</details>

`stuff/lol/loller.py`

<details>

```python
class Loller:

    def __init__(self) -> None:
        pass

    def greeting(self):
        print("Hello from Loller!")

if __name__ == '__main__':
    pass
```

</details>

Here's an image how it looks like on neovim:

<img width="1440" alt="Bildschirmfoto 2021-09-02 um 23 41 46" src="https://user-images.githubusercontent.com/50843046/131920099-44866d10-0a23-4ca7-8d2b-0acfacb7bcb3.png">

As you can see, I'm getting an unresolved import in the `linux.py` file but the import of `linux` "works" (I can go to the Definition of `Base` for example).

**Expected behavior**
A resolved import.

**VS Code extension or command-line**
```
Are you running pyright as a VS Code extension or a command-line tool? Which version? You can find the version of the VS Code extension by clicking on the Pyright icon in the extensions panel.
```

I'm running it on neovim with `nvim-lspconfig`.
This is the pyright version:

```
pyright 1.1.161
```

**Additional context**
I'm aware of that I can't execute the files since they are not in subdirectories. However I've this filestructure

**Comment by TornaxO7** [comment]
This problem needs some more investigation. I just found out that my quick example here works (but not on my project). I'll reopen this if I find something! (Sorry for this unnecessary issue...)

**Comment by erictraut** [comment]
By default, pyright assumes that the top directory of your project is in your import resolution path list. Other directories are not used as search roots unless you configure them as such.

You are importing from `common`, but this directory isn't at the top level of your project, so it can't be resolved.

If your intent is for `stuff` to be used as a root directory for import resolution, you can add it to the `extraPaths` setting in the pyrightconfig.json configuration file.

**Comment by TornaxO7** [comment]
Yeeeeeeeeeah!!! Adding `stuff` to the `extraPaths` in the global configs works!!! Thank you so much for the explanation! :) (and especially for this neat LSP!)
